### ISSUE TITLE
Invalid XML parsing fails when `async` is set to `true`

### ISSUE THREAD

**Issue by fiznool** [Leonidas-from-XIV/node-xml2js#232]
This seems to be a side-effect of #229.

`async` set to `false`:
``` js
var xml2js = require('xml2js');
var parser = new xml2js.Parser({
  async: false,
});

var invalid_xml = '"some data"';
parser.parseString(invalid_xml, function(err, results){
  console.log("ERRORS: " + err);
  console.log("RESULTS: " + results);
});

```
yields (as expected):

```
ERRORS: Error: Non-whitespace before first tag.
Line: 0
Column: 1
Char: "
RESULTS: undefined
```

However, `async` set to `true`:

``` js
var xml2js = require('xml2js');
var parser = new xml2js.Parser({
  async: true,
});

var invalid_xml = '"some data"';
parser.parseString(invalid_xml, function(err, results){
  console.log("ERRORS: " + err);
  console.log("RESULTS: " + results);
});

```

throws an unhandled exception:

```
ERRORS: Error: Non-whitespace before first tag.
Line: 0
Column: 1
Char: "
RESULTS: undefined
events.js:85
      throw er; // Unhandled 'error' event
            ^
Error: Text data outside of root node.
Line: 0
Column: 2
Char: s
    at error (/home/fiznool/Code/sandbox/xml2js-parse-exception/node_modules/xml2js/node_modules/sax/lib/sax.js:652:8)
    at strictFail (/home/fiznool/Code/sandbox/xml2js-parse-exception/node_modules/xml2js/node_modules/sax/lib/sax.js:675:22)
    at Object.write (/home/fiznool/Code/sandbox/xml2js-parse-exception/node_modules/xml2js/node_modules/sax/lib/sax.js:981:13)
    at Parser.exports.Parser.Parser.processAsync (/home/fiznool/Code/sandbox/xml2js-parse-exception/node_modules/xml2js/lib/xml2js.js:254:41)
    at Immediate._onImmediate (/home/fiznool/Code/sandbox/xml2js-parse-exception/node_modules/xml2js/lib/xml2js.js:7:59)
    at processImmediate [as _immediateCallback] (timers.js:367:17)
```

I'm guessing this has something to do with the `processAsync` function causing the error event to be emitted in a new cycle, which isn't caught by the error handler setup in `parseString`. But I'm not really sure.

**Comment by tflanagan** [comment]
This is fixed in #240

**Comment by Leonidas-from-XIV** [comment]
Merged #240.
